### PR TITLE
rbd-mirror: image status should report remote status

### PR DESCRIPTION
**Pull request by dillaman**
TODO
- [x] push status from `rbd-mirror` daemon to remote peer
- [x] update CLI to show remote peer status
- [x] fix teuthology tests to handle new status output

### CODE REVIEW

**Comment by dillaman** [comment]
> @dillaman nit: in "cls/rbd: preliminary support for mirror image status from multiple sites" commit log message when you refer `mirror_image_set` and `mirror_image_get/list` methods I suppose you meant `mirror_image_status_set` and `mirror_image_status_get/list`?

Indeed -- tweaked.

**Comment by tchaikov** [comment]
jenkins test make check

File: src/cls/rbd/cls_rbd.cc [hunk 1]
```diff
@@ -4732,20 +4743,192 @@ int read_peer(cls_method_context_t hctx, const std::string &id,
...
+  for (auto const &peer : peers) {
+    if (peer.uuid == mirror_peer.uuid) {
+      CLS_ERR("peer uuid '%s' already exists",
+              peer.uuid.c_str());
+      return -ESTALE;
+    } else if (peer.site_name == mirror_peer.site_name) {
+      CLS_ERR("peer site name '%s' already exists",
+              peer.site_name.c_str());
+      return -EEXIST;
+    } else if (!mirror_peer.fsid.empty() && peer.fsid == mirror_peer.fsid) {
```
Comment (tchaikov) [inline src/cls/rbd/cls_rbd.cc]: @trociny and @dillaman this change breaks the upgrade test. as we don't check pool_id anymore. see https://tracker.ceph.com/issues/42384

Comment (trociny) [reply src/cls/rbd/cls_rbd.cc]: @tchaikov I believe it is fixed by #30948

Comment (tchaikov) [reply src/cls/rbd/cls_rbd.cc]: thanks @trociny ! Jason marked the ticket as a dup.

Comment (trociny) [reply src/cls/rbd/cls_rbd.cc]: Ah, @dillaman has already marked it as a duplicate.

**Pull request merged**